src/lib/state/reducers/results.js

```diff
@@ -21,29 +21,38 @@ export default (state = {}, action) => {
         data: {
           ...state.data,
         },
+        queryState: action.payload.queryState,
       };
     case RESULTS_FETCH_SUCCESS:
-      return {
-        loading: false,
-        data: {
-          ...state.data,
-          aggregations: action.payload.aggregations,
-          hits: action.payload.hits,
-          total: action.payload.total,
-        },
-        error: {},
-      };
+      if (state.queryState === action.payload.queryState) {
+        return {
+          loading: false,
+          data: {
+            ...state.data,
+            aggregations: action.payload.aggregations,
+            hits: action.payload.hits,
+            total: action.payload.total,
+          },
+          error: {},
+        };
+      } else {
+        return state;
+      }
     case RESULTS_FETCH_ERROR:
-      return {
-        loading: false,
-        data: {
-          ...state.data,
-          aggregations: {},
-          hits: [],
-          total: 0,
-        },
-        error: action.payload,
-      };
+      if (state.queryState === action.payload.queryState) {
```

Comment:
Maybe this can be done in the action level so that the reducer function stays pure :)

Reply:
I'm not sure if it will work because we need to know the current state in order to update it accordingly.

Reply:
same as above